ts/Core/Chart/Chart.ts

```diff
@@ -2414,106 +2412,102 @@ class Chart {
 
         let correction = 0; // correction for X axis labels
 
-        // Title
-        chart.setTitle();
+        fireEvent(chart, 'rendering', { args: arguments }, function (): void {
 
-        /**
-         * The overview of the chart's series.
-         *
-         * @name Highcharts.Chart#legend
-         * @type {Highcharts.Legend}
-         */
-        chart.legend = new Legend(chart, options.legend as any);
-
-        // Get stacks
-        if (chart.getStacks) {
-            chart.getStacks();
-        }
+            // Title
+            chart.setTitle();
```

Comment:
Here is is called without Legend, @TorsteinHonsi .

Reply:
So it's called _twice_ when the Legend is imported? 🤔

Reply:
Yes, correct.

Reply:
Okay, so the reason for this is that we wanted to wrap the _whole_ rendering logic inside one event handler, and this `setTitle` call was done before the legend, right?

If so, I think we should rather create an event call after setting the title but before `getMargins()`, where we could hook in the legend creation. 

In this PR, there's also an `afterRendering` event that is not used by any component.

Reply:
> Okay, so the reason for this is that we wanted to wrap the _whole_ rendering logic inside one event handler, and this `setTitle` call was done before the legend, right?

Yes, and it is needed. We tested with changed position and it started to fail.

> If so, I think we should rather create an event call after setting the title but before `getMargins()`, where we could hook in the legend creation.

Do you mean that we move `setTitle` in front of the default handler, or do we have no default handler at all?

> In this PR, there's also an `afterRendering` event that is not used by any component.

`render` and `afterRender` are hijacked by redraw and update. Therefor we had to go with `rendering` / `afterRendering`.

Reply:
> do we have no default handler at all?

Yes, that's what I think. In this case we should have a detailed event that is called at just the right moment. Like we do for example with `beforeRender`

https://github.com/highcharts/highcharts/blob/a905c72df5599de95cf9a8fd040d59338dde530c/ts/Core/Chart/Chart.ts#L2707-L2711